tests/models/gemma2/test_modeling_gemma2.py

```diff
@@ -113,6 +113,12 @@ def test_prompt_lookup_decoding_matches_greedy_search(self, assistant_type):
     def test_assisted_decoding_sample(self):
         pass
 
+    @parameterized.expand([(None, True), ("static", False)])
+    def test_assisted_decoding_with_num_logits_to_keep(self, cache_implementation, return_legacy_cache):
+        if cache_implementation == "static":
+            self.skipTest("Gemma2 has HybridCache which is not compatible with assisted decoding StaticCache")
+            pass
+
```

Comment:
let's not skip entirely, but only the `static_cache` test, as we still need to check if assisted generation works in Gemma2 :)

Maybe it will be skipped by the `model._support_static_cache` as I've commented above, but if not we can skip only the `test_assisted_decoding_with_num_logits_to_keep_1_static` (maybe it's called a bit differently)

Reply:
i switch to `_supports_static_cache` to skip the case. For Gemma, it's a bit different, since it's using HybridCache and claims `_supports_static_cache = True`, I still skip it in model test file. Will remove this skip after enable HybridCache for assisted decoding, I plan to enable it after this PR(pure StaticCache) merged, thx.